### PR TITLE
Using nextPageToken to fetch next page data from repository API call (PROJQUAY-5237)

### PR DESCRIPTION
**Pull request by Sunandadadi**

### CODE REVIEW

File: src/hooks/UseRepositories.ts [hunk 1]
```diff
@@ -13,35 +13,72 @@ export function useRepositories(organization?: string) {
...
     isPlaceholderData,
     error,
-  } = useQuery(
-    ['organization', organization, 'repositories'],
-    currentOrganization
-      ? ({signal}) => fetchRepositoriesForNamespace(currentOrganization, signal)
-      : ({signal}) => fetchAllRepos(listOfOrgNames, true, signal),
-    {
-      placeholderData: [],
+  } = useQuery({
```
Comment (bcaton85) [inline src/hooks/UseRepositories.ts:39]: It seems there are two paging approaches conflated in this implementation. As of now when clicking through the pages in the `projectquay` organization all 150 repos are being appended to the total list. 
image
Gathering all the repos up front has already been implemented in RepositoryResource.ts:L81. The changes here should probably be reverted in favor of the other approach.

**Review by bcaton85 — CHANGES_REQUESTED** [review]

File: src/hooks/UseRepositories.ts (hunk 1 above)
Comment (Sunandadadi) [reply src/hooks/UseRepositories.ts:39]: Realised that [RepositoryResource.ts:L81.](https://github.com/quay/quay-ui/pull/166/files#diff-281eaf07bf2aa85cb2f5c77cb7478e221613871ced40e342bfb6e872b5d574feR81) actually needs to be an if condition. I'd need the `onSuccess` function to handle the different types of responses returned.

**Review by bcaton85 — CHANGES_REQUESTED** [review]

File: src/routes/RepositoryDetails/RepositoryDetails.tsx [hunk 2]
```diff
@@ -57,6 +57,7 @@ export default function RepositoryDetails() {
   const [drawerContent, setDrawerContent] = useState<DrawerContentType>(
     DrawerContentType.None,
   );
+  const [repoDetails, setRepoDetails] = useState<IRepositoryDetails>();
```
[the remark this answers is not in the document]
Comment (Sunandadadi) [reply src/routes/RepositoryDetails/RepositoryDetails.tsx:60]: Agreed, removing it given it is never used.

File: src/hooks/UseRepositories.ts [hunk 3]
```diff
@@ -25,23 +26,28 @@ export function useRepositories(organization?: string) {
...
+    queryKey: ['organization', organization || 'all', 'repositories', page],
+    keepPreviousData: true,
+    placeholderData: [],
+    queryFn: ({signal}) => {
+      return currentOrganization
+        ? fetchRepositoriesForNamespace(
+            currentOrganization,
+            signal,
+            nextPageToken,
```
[the remark this answers is not in the document]
Comment (Sunandadadi) [reply src/hooks/UseRepositories.ts]: Yes, agreed, removed the state.

**Review by bcaton85 — APPROVED** [review]
LGTM

**Pull request merged**